abi-versions/v0.2.1/README.md

```diff
@@ -0,0 +1,2166 @@
+# Proxy-Wasm ABI v0.2.1 specification
+
+---
+
+Status: **FINAL** (corrections and/or clarifications are welcomed)
+
+---
+
+# Terminology
+
+- *Plugin* - a WebAssembly module implementing Proxy-Wasm ABI. There are
+  3 types of plugins: *HTTP*, *Stream* (TCP) and *Background task*.
+
+- *Host* - a proxy or another environment that executes Proxy-Wasm Plugin
+  inside the WebAssembly Virtual Machine (WasmVM).
+
+- *Hostcall* - a function call from Plugin to Host.
+
+- *Callback* - a function call from Host to Plugin (usually, as a result
+  of an external event).
+
+- *Plugin Context* (also known as *Root Context*) - a context in which
```

Comment:
Might help to explain how they relate to the execution context. E.g. whether the same VM is called for many streams or not.

Reply:
Let's leave the lifecycle documentation for a separate PR or document.